src/token/components/TextInput.tsx

```diff
@@ -0,0 +1,48 @@
+import { CSSProperties, ChangeEvent, useCallback } from 'react';
+import { SearchIcon } from './SearchIcon';
+
+const styles = {
+  container: {
+    position: 'relative',
+  },
+  icon: {
+    position: 'absolute',
+    left: '16px',
+    top: '14px',
+  },
+  input: {
+    borderRadius: '1000px',
+    padding: '8px 20px 8px 48px',
+    width: '100%',
+    background: '#EEF0F3',
+    color: '#5B616E',
+  },
+} as Record<string, CSSProperties>;
+
+type TextInputReact = {
+  placeholder: string;
+  value: string;
+  onChange: (value: string) => void;
+};
+
+export function TextInput({ placeholder, value, onChange }: TextInputReact) {
```

Comment:
are we protected over XSS vulnerability on this?

Reply:
For this specific instance with TokenSelector, it should be safe. But I haven't thought about XSS attacks so I will dig into that deeper.